src/main/java/org/web3j/codegen/SolidityFunctionWrapperGenerator.java

```diff
@@ -432,37 +439,99 @@ private static void buildTransactionFunction(
         methodBuilder.addStatement("return executeTransactionAsync(function)");
     }
 
-    private static MethodSpec buildEventFunction(
-            AbiDefinition functionDefinition) throws ClassNotFoundException {
+    static TypeSpec buildEventResponseObject(String className, List<NamedTypeName> indexedParameters, List<NamedTypeName> nonIndexedParameters) {
+        TypeSpec.Builder builder = TypeSpec.classBuilder(className).addModifiers(Modifier.PUBLIC, Modifier.STATIC);
+        for (NamedTypeName namedType : indexedParameters) {
+            builder.addField(namedType.getTypeName(), namedType.getName(), Modifier.PUBLIC);
+        }
 
-        String functionName = functionDefinition.getName();
-        String generatedFunctionName =
-                "process" + Strings.capitaliseFirstLetter(functionName) + "Event";
+        for (NamedTypeName namedType : nonIndexedParameters) {
+            builder.addField(namedType.getTypeName(), namedType.getName(), Modifier.PUBLIC);
+        }
+
+        return builder.build();
+    }
+
+    static MethodSpec buildEventObservableFunction(String responseClassName, String functionName, List<NamedTypeName> indexedParameters, List<NamedTypeName> nonIndexedParameters) throws ClassNotFoundException {
```

Comment:
Please can you ensure you keep lines to less than 100 chars in length. I haven't added checkstyle constraints just yet.